index.d.ts

```diff
@@ -0,0 +1,456 @@
+declare module 'oidc-provider' {
+    import { Request, Response, Handler } from 'express';
+    import Koa = require('koa');
+    import Router = require('koa-router');
+    import { IncomingMessage, ServerResponse } from 'http';
+    import EventEmitter = require('events');
+    import { JWK } from 'node-jose';
+
+    type SubjectType = 'public' | 'parwise';
+
+    interface IClient {
+        applicationType?: 'web' | 'native';
+        grant_types?: string[];
+        id_token_signed_response_alg?: string;
+        require_auth_time?: boolean;
+        response_types?: string[];
+        subject_type?: SubjectType;
+        token_endpoint_auth_method?: string;
```

Comment:
defaults to `client_secret_basic` and there's an enum list for the supported values. dtto for introspection and revocation counterparts